packages/cli/src/services/core/components/SetMinecraftVersion.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect } from "react"
+import SelectInput, { Item, ItemProps } from "ink-select-input"
+import { useState } from "react"
+import { checkForAssets, detectVersions } from "../../../utils"
+import { Box, Text } from "ink"
+import { CACHE } from "../../../main"
+import { MinecraftUtility } from "../../minecraft"
+
+const minecraftAssetReader = new MinecraftUtility()
+
+type ChoiceValue = any
+interface ChoiceOption {
+  key?: string
+  label: string
+  value: ChoiceValue
+}
+export interface Item<V> {
+  key?: string
+  label: string
+  value: V
+}
+
+export const SetMinecraftVersion = (props: {
+  clearSelectedOptionHandler: () => void
+  setRawAssetsPathHandler: (v: string) => void
+}) => {
+  const [selectedOption, setSelectedOption] = useState(
+    (null as unknown) as string
+  )
+  const clearSelectedOptionHandler = () =>
+    setSelectedOption((null as unknown) as string)
+  const [minecraftVersions, setMinecraftVersions] = useState()
+  const [selectedVersion, setSelectedVersion] = useState(``)
+
+  const selectHandler = (value: any) => {
+    setSelectedVersion(value)
+    if (value) {
+      props.clearSelectedOptionHandler()
+    }
+  }
+
+  // TODO: Find out why there's a memory leak in this component. It's probably in useEffect().
+  useEffect(() => {
+    let isMounted = true
+    let minecraftVersionsArray: any
+
+    detectVersions()
```

Comment:
Think I found your memory leak! This asynchronous code will happen out-of-sync from the `checkForAssets` code. Currently, the `detectVersions()` _will_ get called, but you can't guarantee that the stuff in the first promise chain (`then` calls) have run by the time the `checkForAssets` code runs.

To fix this, you could simply add another `.then()` to the end of this promise chain and add the `checkForAssets` code to that:
```js
//..
.then(() => checkForAssets(selectedVersion, { clearSelectedOptionHandler })
.then(
      (path) => {
        if (path) {
          minecraftAssetReader.readInRawData({
            path,
          })
          CACHE.setRootAssetsPath(path)
          props.setRawAssetsPathHandler(path)
        } else {
          console.error(`path did not exist when passed to readInRawData`)
        }
      }
    )
//...
```

Reply:
I tried this and it worked! :D